src/main/kotlin/uk/gov/justice/digital/hmpps/authorizationserver/resource/ClientDeploymentController.kt

```diff
@@ -29,6 +31,22 @@ class ClientDeploymentController(
       "username" to authenticationFacade.currentUsername!!,
       "baseClientId" to clientIdService.toBase(clientDeployment.clientId),
     )
+    telemetryClient.trackEvent("AuthorizationServerClientDeploymentDetailsAdded", telemetryMap)
+  }
+
+  @PutMapping("clients/deployment/{clientId}")
```

Comment:
Sorry @ravmoj actually I think it might be better as 

POST /clients/<CLIENT ID>/deployment
PUT /clients/<CLIENT ID>/deployment

so the only difference between the two endpoints should be the verb POST or PUT.  That way we can remove the client id from the request object entirely.  Don't think there's anything sensitive that we'd not want in the URL.

Reply:
Updated as discussed